code/src/utils/createSkeletonCards.tsx

```diff
@@ -0,0 +1,9 @@
+import ProductSkeleton from "@/components/product-skeleton/ProductSkeleton";
+
+const createProductSkeletons = (iterations: number) => {
+  return Array.from({ length: iterations }, (_, index) => (
+    <ProductSkeleton key={index} />
+  ));
+};
+
+export default createProductSkeletons;
```

Comment:
Please try to accept component property to render any kind of elements, not just `ProductSkeleton`. Also don't forget to write tests

Reply:
Fixed the AppSkeleton thing and created unit test for the function. I decided to keep the function working with only ProductSkeleton components for now.